### PR TITLE
Rename GetWorkflowTables -> GetWorkflowObjects

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
- Rename tables to objects because the things written to by the workflow in the integration are not necessarily tables.
- Update `integration_tests/backend/test_reads.py` to delete workflows & wait for flow runs in a similar way as in the SDK.
## Related issue number (if any)
ENG 1394 (builds on this PR)
ENG-1369 (builds on above)
ENG-1325 (builds on above)
ENG-1431 (builds on above)

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [N/A] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [N/A] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Review by likawind — COMMENTED** [review]
Since we plan to use 'SavedObjects' in sdk, should we use `SavedObject` here as well?

**Review by likawind — COMMENTED** [review]
We could keep `/objects` for the api route based on Chenggang's comment in https://github.com/aqueducthq/aqueduct/pull/240#pullrequestreview-1048401809

File: integration_tests/backend/test_reads.py [hunk 1]
```diff
@@ -5,12 +5,13 @@
...
 import requests
+import utils
 
 import aqueduct
 
 
-class TestBackend:
-    GET_WORKFLOW_TABLES_TEMPLATE = "/api/workflow/%s/tables"
+class TestReads:
+    LIST_WORKFLOW_TABLES_TEMPLATE = "/api/workflow/%s/objects"
```
Comment (likawind) [inline integration_tests/backend/test_reads.py:14]: nit: LIST_WORKFLOW_SAVED_OBJECTS_TEMPLATE

**Review by likawind — COMMENTED** [review]
@eunice-chan have you decided to use 'Object' or 'SavedObject' here?

File: integration_tests/backend/setup/changing_saves.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 table = integration.sql(query="SELECT * FROM wine;")
 
-table.save(integration.config(table="table_1", update_mode="append"))
+table.save(integration.config(table="table_1", update_mode="replace"))
```
Comment (eunice-chan) [inline integration_tests/backend/setup/changing_saves.py:22]: Update this so we aren't appending to a non-existing table and thus these flows fail

File: integration_tests/backend/utils.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        statuses = [flow_run["status"] for flow_run in flow_runs]
+
+        # Continue checking as long as there are still runs pending.
+        if any(status == ExecutionStatus.PENDING for status in statuses):
+            continue
+
+        if len(flow_runs) < num_runs:
+            continue
+
+        if expect_success is not None:
```
Comment (eunice-chan) [inline integration_tests/backend/utils.py:44]: This & `expect_success` default value is the only change from the SDK integration test utils. Only requiring execution of each flow run to not be pending.

**Review by likawind — APPROVED** [review]

**Pull request merged**